File: backend/apps/bug_note/models/defect.py
from django.db import models
from apps.utility.models import BaseModel
from apps.staff_hub.models.user import User
from apps.trade_flow.models.orders import Order


class Defect(BaseModel):
    create_user = models.ForeignKey(User, on_delete=models.CASCADE, verbose_name="作成者")
    order = models.ForeignKey(Order, on_delete=models.CASCADE, verbose_name="注文")
    occurred_at = models.DateTimeField("発生日時")
    title = models.CharField("タイトル", max_length=255)
    defect_detail = models.CharField("不具合詳細", max_length=2000)
    submission = models.CharField("対策", max_length=2000)
    submission_deadline = models.DateTimeField("対策期限")

    class Meta:
        verbose_name = "不具合連絡"
        verbose_name_plural = "不具合連絡"
        db_table = "defects"

    def __str__(self):
        return self.title

    @classmethod
    def get_defects_by_month(cls):
        """
        不具合連絡を一括で全て取得する
        """
        return cls.objects.select_related("order", "create_user").filter(deleted_at__isnull=True)

File: frontend/src/fixtures/defect.ts
import { DefectTableList, DefectItem } from "@/types/defect";

export const mockDefectTableList: DefectTableList[] = [
  {
    id: 1,
    create_user_name: "作成者1",
    order_number: "注文番号1",
    occurred_at: "2021-01-01",
    title: "タイトル1",
    defect_detail_button: "",
  },
  {
    id: 2,
    create_user_name: "作成者2",
    order_number: "注文番号2",
    occurred_at: "2021-01-02",
    title: "タイトル2",
    defect_detail_button: "",
  },
  {
    id: 3,
    create_user_name: "作成者3",
    order_number: "注文番号3",
    occurred_at: "2021-01-03",
    title: "タイトル3",
    defect_detail_button: "",
  },
];

export const mockDefectItem: DefectItem = {
  id: 1,
  title: "タイトル1",
  defect_detail: "不具合詳細1",
  submission: "対策1",
  submission_deadline: "2021-01-01",
  create_user: 1,
  order: 1,
  created_at: "2021-01-01",
  updated_at: "2021-01-01",
  occurred_at: "2021-01-01",
};
